modules/porous_flow/src/materials/PorousFlowJ2.C

```diff
@@ -0,0 +1,91 @@
+  params.addClassDescription("This Material forms a std::vector of properties, old properties (optionally), and derivatives, out of the individual phase properties");
+  return params;
+}
+
+PorousFlowJ2::PorousFlowJ2(const InputParameters & parameters) :
+    DerivativeMaterialInterface<Material>(parameters),
+
+    _dictator_UO(getUserObject<PorousFlowDictator>("PorousFlowDictator_UO")),
+    _pressure_variable_name(_dictator_UO.pressureVariableNameDummy()),
+    _saturation_variable_name(_dictator_UO.saturationVariableNameDummy()),
+    _temperature_variable_name(_dictator_UO.temperatureVariableNameDummy()),
+    _mass_fraction_variable_name(_dictator_UO.massFractionVariableNameDummy()),
+    _num_phases(_dictator_UO.numPhases()),
+    _num_var(_dictator_UO.numVariables()),
+    _pf_prop(getParam<std::string>("material_property")),
+    _include_old(getParam<bool>("include_old")),
+    _at_qps(getParam<bool>("position")),
+
+    _dporepressure_dvar(_at_qps ? getMaterialProperty<std::vector<std::vector<Real> > >("dPorousFlow_porepressure_qp_dvar") : getMaterialProperty<std::vector<std::vector<Real> > >("dPorousFlow_porepressure_nodal_dvar")),
+    _dsaturation_dvar(_at_qps ? getMaterialProperty<std::vector<std::vector<Real> > >("dPorousFlow_saturation_qp_dvar") : getMaterialProperty<std::vector<std::vector<Real> > >("dPorousFlow_saturation_nodal_dvar")),
+    _dtemperature_dvar(_at_qps ? getMaterialProperty<std::vector<std::vector<Real> > >("dPorousFlow_temperature_qp_dvar") : getMaterialProperty<std::vector<std::vector<Real> > >("dPorousFlow_temperature_nodal_dvar")),
+
+    _property(declareProperty<std::vector<Real> >(_pf_prop)),
+    _property_old(_include_old ? &declarePropertyOld<std::vector<Real> >(_pf_prop) : NULL),
+    _dproperty_dvar(declareProperty<std::vector<std::vector<Real> > >("d" + _pf_prop + "_dvar"))
+{
+  _phase_property.resize(_num_phases);
+  for (unsigned int ph = 0; ph < _num_phases; ++ph)
+    _phase_property[ph] = &getMaterialProperty<Real>(_pf_prop + Moose::stringify(ph));
+
+    _dphase_property_dp.resize(_num_phases);
+    _dphase_property_ds.resize(_num_phases);
+    _dphase_property_dt.resize(_num_phases);
+
+  for (unsigned int ph = 0; ph < _num_phases; ++ph)
+  {
+    _dphase_property_dp[ph] = &getMaterialPropertyDerivative<Real>(_pf_prop + Moose::stringify(ph), _pressure_variable_name);
+    _dphase_property_ds[ph] = &getMaterialPropertyDerivative<Real>(_pf_prop + Moose::stringify(ph), _saturation_variable_name);
+    _dphase_property_dt[ph] = &getMaterialPropertyDerivative<Real>(_pf_prop + Moose::stringify(ph), _temperature_variable_name);
+  }
+}
+
+void
+PorousFlowJ2::initQpStatefulProperties()
+{
+  _property[_qp].resize(_num_phases);
+  for (unsigned int ph = 0; ph < _num_phases; ++ph)
+    _property[_qp][ph] = (*_phase_property[ph])[_qp];
+
+  if (_include_old)
+    _property_old[_qp].resize(_num_phases);
```

Comment:
This shouldn't be necessary. The `initQpStatefulProperties` method shouldn't need to touch the old and older properties at all. We size and populate the old and older properties with those in the "current" properties after this call. Have you seen behavior to the contrary, or do we just need to improve our doco?